Jenkinsfile

```diff
@@ -28,87 +28,90 @@ void buildManifest(String variant_name, String bitbake_image) {
-                    vagrant("rsync -trpg ${yoctoDir}/build/downloads/ ${env.YOCTO_CACHE_ARCHIVE_PATH}/downloads/")
-                    vagrant("rsync -trpg ${yoctoDir}/build/sstate-cache/ ${env.YOCTO_CACHE_ARCHIVE_PATH}/sstate-cache")
+            stage("Fetchall ${variant_name}") {
+                // Without cache access, we do want to do fetchall, but only then.
+                if (!env.YOCTO_CACHE_URL?.trim()) {
+                    vagrant("/vagrant/cookbook/yocto/fetch-sources-for-recipes.sh ${yoctoDir} ${bitbake_image}")
                 }
             }
-        }
 
-        if (env.NIGHTLY_BUILD) {
-            stage("Archive artifacts ${variant_name}") {
-                String artifactDir = "artifacts_${variant_name}"
+            try {
+
+                stage("Bitbake ${variant_name}") {
+                    vagrant("/vagrant/cookbook/yocto/build-images.sh ${yoctoDir} ${bitbake_image}")
+                    vagrant("/vagrant/cookbook/yocto/build-images.sh ${yoctoDir} ${bitbake_image}-dev")
+                }
+
+                stage("Build SDK ${variant_name}") {
+                    vagrant("/vagrant/cookbook/yocto/build-sdk.sh ${yoctoDir} ${bitbake_image}")
+                    vagrant("/vagrant/cookbook/yocto/build-sdk.sh ${yoctoDir} ${bitbake_image}-dev")
+                }
+            } finally {
+                stage("Archive cache ${variant_name}") {
+                    // Archive the downloads and sstate when the environment variable was set to true
+                    // by the Jenkins job.
+                    if (env.ARCHIVE_CACHE && env.YOCTO_CACHE_ARCHIVE_PATH?.trim()) {
+                        vagrant("rsync -trpg ${yoctoDir}/build/downloads/ ${env.YOCTO_CACHE_ARCHIVE_PATH}/downloads/")
+                        vagrant("rsync -trpg ${yoctoDir}/build/sstate-cache/ ${env.YOCTO_CACHE_ARCHIVE_PATH}/sstate-cache")
+                    }
+                }
+
+                if (env.NIGHTLY_BUILD) {
+                    stage("Archive artifacts ${variant_name}") {
+                        String artifactDir = "artifacts_${variant_name}"
 
-                sh "rm -rf ${artifactDir}"
-                sh "mkdir ${artifactDir}"
+                        sh "rm -rf ${artifactDir}"
+                        sh "mkdir ${artifactDir}"
 
-                // Copy images and SDK to the synced directory
-                vagrant("/vagrant/ci-scripts/copy_to_archive ${yoctoDir}/build /vagrant/${artifactDir}")
+                        // Copy images and SDK to the synced directory
+                        vagrant("/vagrant/ci-scripts/copy_to_archive ${yoctoDir}/build /vagrant/${artifactDir}")
 
-                // And save them in Jenkins
-                archiveArtifacts "${artifactDir}/**"
+                        // And save them in Jenkins
+                        archiveArtifacts "${artifactDir}/**"
+                    }
+                }
+            }
+        } finally { // Archive cache even if there were errors.
```

Comment:
The code down there doesn't archive anything it just destroys the vagrand machine, I'm not sure this comment makes sense here.

Reply:
No, the comment is out of place. Will update!